### PR TITLE
fix: Replace Promise.allSettled by RSVP.allSettled

### LINKED ISSUE #4401: Replace Promise.allSettled by RSVP.allSettled
Promise.allSettled is not polyfilled, replace with RSVP.allSettled

Parent: #4181

### PR DESCRIPTION
**Pull request by EmperorYP7**
Fixes #4401

#### Short description of what this resolves:
Attempted to add polyfilled function to the relevant files to enhance compatibility across browsers.

#### Changes proposed in this pull request:

-Replacing Promise.allSettled() function with RSVP.allSettled() function in the app files.

#### Checklist

- [X] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [X] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: app/controllers/public/index.js [hunk 1]
```diff
@@ -135,11 +136,10 @@ export default class IndexController extends Controller {
         })
         .catch(async e => {
           console.error('Error while saving order', e);
-          try {
-            await Promise.allSettled((attendees ? attendees.toArray() : []).map(attendee => attendee.destroyRecord()));
-          } catch (error) {
-            console.error('Error while deleting attendees after order failure', error);
-          }
+          await RSVP.allSettled(attendees ? attendees.toArray() : []).then(attendee => attendee.destroyRecord(),
+            (error) => {
```
Comment (niranjan94) [inline app/controllers/public/index.js:140]: Codacy Issue found: [Unexpected parentheses around single function argument.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5583663)

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 10
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5583663&bid=18300760)

File: app/mixins/event-wizard.js [hunk 2]
```diff
@@ -41,23 +42,20 @@ export default Mixin.create(MutableArray, CustomFormMixin, {
   },
 
   /**
-   * Save event & related data
-   *
+   * Save event & related data 
+   *
```
Comment (niranjan94) [inline app/mixins/event-wizard.js:46]: Codacy Issue found: [Trailing spaces not allowed.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5583663)

File: app/mixins/event-wizard.js [hunk 3]
```diff
@@ -41,23 +42,20 @@ export default Mixin.create(MutableArray, CustomFormMixin, {
...
-    const results = await Promise.allSettled(propsToSave.map(property => {
-      try {
-        return event.get(property);
-      } catch (e) {
-        if (!(e.errors && e.errors.length && e.errors.length > 0 && e.errors[0].status === 404)) {
-          // Lets just ignore any 404s that might occur. And throw the rest for the caller fn to catch
+    const results = await RSVP.allSettled(propsToSave).then(property => {
+      return event.get(property);
+    }, (e) => {
+        if (!(e.errors && e.errors.length && e.errors.length > 0 && e.errors[0].status === 404))
```
Comment (niranjan94) [inline app/mixins/event-wizard.js:55]: Codacy Issue found: [Expected { after 'if' condition.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5583663)

File: app/mixins/event-wizard.js [hunk 4]
```diff
@@ -41,23 +42,20 @@ export default Mixin.create(MutableArray, CustomFormMixin, {
   },
 
   /**
-   * Save event & related data
-   *
+   * Save event & related data
```
Comment (niranjan94) [inline app/mixins/event-wizard.js:45]: Codacy Issue found: [Trailing spaces not allowed.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5583663)

File: app/controllers/public/index.js [hunk 5]
```diff
@@ -135,11 +136,10 @@ export default class IndexController extends Controller {
         })
         .catch(async e => {
           console.error('Error while saving order', e);
-          try {
-            await Promise.allSettled((attendees ? attendees.toArray() : []).map(attendee => attendee.destroyRecord()));
-          } catch (error) {
-            console.error('Error while deleting attendees after order failure', error);
-          }
+          await RSVP.allSettled(attendees ? attendees.toArray() : []).then(attendee => attendee.destroyRecord(),
```
Comment (niranjan94) [inline app/controllers/public/index.js:139]: Codacy Issue found: ['RSVP' is not defined.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5583663)

File: app/mixins/event-wizard.js (hunk 3 above)
Comment (niranjan94) [inline app/mixins/event-wizard.js:55]: Codacy Issue found: [Expected indentation of 6 spaces but found 8.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5583663)

File: app/mixins/event-wizard.js [hunk 6]
```diff
@@ -3,6 +3,7 @@ import MutableArray from '@ember/array/mutable';
 import RSVP from 'rsvp';
 import { v1 } from 'ember-uuid';
 import CustomFormMixin from 'open-event-frontend/mixins/custom-form';
+import { allSettled } from 'rsvp';
```
Comment (niranjan94) [inline app/mixins/event-wizard.js:6]: Codacy Issue found: ['allSettled' is defined but never used.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5583663)

Comment (niranjan94) [inline app/mixins/event-wizard.js:6]: Codacy Issue found: ['rsvp' import is duplicated.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5583663)

File: app/controllers/public/index.js [hunk 7]
```diff
@@ -2,6 +2,7 @@ import classic from 'ember-classic-decorator';
 import { action, computed } from '@ember/object';
 import Controller from '@ember/controller';
 import { htmlSafe } from '@ember/string';
+import { allSettled } from 'rsvp';
```
Comment (niranjan94) [inline app/controllers/public/index.js:5]: Codacy Issue found: ['allSettled' is defined but never used.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5583663)

File: app/mixins/event-wizard.js [hunk 8]
```diff
@@ -41,23 +42,20 @@ export default Mixin.create(MutableArray, CustomFormMixin, {
...
     const data = {};
-    const results = await Promise.allSettled(propsToSave.map(property => {
-      try {
-        return event.get(property);
-      } catch (e) {
-        if (!(e.errors && e.errors.length && e.errors.length > 0 && e.errors[0].status === 404)) {
-          // Lets just ignore any 404s that might occur. And throw the rest for the caller fn to catch
+    const results = await RSVP.allSettled(propsToSave).then(property => {
+      return event.get(property);
+    }, (e) => {
```
Comment (niranjan94) [inline app/mixins/event-wizard.js:54]: Codacy Issue found: [Unexpected parentheses around single function argument.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5583663)

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 7
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5583663&bid=18300760)

File: app/mixins/event-wizard.js (hunk 6 above)
Comment (niranjan94) [inline app/mixins/event-wizard.js:6]: Codacy Issue found: ['rsvp' import is duplicated.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5583663)

Comment (niranjan94) [inline app/mixins/event-wizard.js:6]: Codacy Issue found: ['allSettled' is defined but never used.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5583663)

File: app/controllers/public/index.js [hunk 9]
```diff
@@ -135,11 +136,9 @@ export default class IndexController extends Controller {
         })
         .catch(async e => {
           console.error('Error while saving order', e);
-          try {
-            await Promise.allSettled((attendees ? attendees.toArray() : []).map(attendee => attendee.destroyRecord()));
-          } catch (error) {
-            console.error('Error while deleting attendees after order failure', error);
-          }
+          await RSVP.allSettled(attendees ? attendees.toArray() : []).then(attendee => attendee.destroyRecord(),
```
Comment (niranjan94) [inline app/controllers/public/index.js:139]: Codacy Issue found: ['RSVP' is not defined.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5583663)

File: app/mixins/event-wizard.js [hunk 10]
```diff
@@ -48,16 +49,14 @@ export default Mixin.create(MutableArray, CustomFormMixin, {
...
     const { event } = this.model;
     const data = {};
-    const results = await Promise.allSettled(propsToSave.map(property => {
-      try {
-        return event.get(property);
-      } catch (e) {
-        if (!(e.errors && e.errors.length && e.errors.length > 0 && e.errors[0].status === 404)) {
+    const results = await RSVP.allSettled(propsToSave).then(property => {
+      return event.get(property);
+    }, (e) => {
```
Comment (niranjan94) [inline app/mixins/event-wizard.js:54]: Codacy Issue found: [Unexpected parentheses around single function argument.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5583663)

File: app/controllers/public/index.js [hunk 11]
```diff
@@ -135,11 +136,9 @@ export default class IndexController extends Controller {
...
+            (error) => console.error('Error while deleting attendees after order failure', error)
```
Comment (niranjan94) [inline app/controllers/public/index.js:140]: Codacy Issue found: [Unexpected parentheses around single function argument.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5583663)

File: app/controllers/public/index.js (hunk 7 above)
Comment (niranjan94) [inline app/controllers/public/index.js:5]: Codacy Issue found: ['allSettled' is defined but never used.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5583663)

File: app/controllers/public/index.js [hunk 12]
```diff
@@ -135,11 +136,9 @@ export default class IndexController extends Controller {
...
+            );
```
Comment (niranjan94) [inline app/controllers/public/index.js:141]: Codacy Issue found: [Expected indentation of 10 spaces but found 12.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5583663)

File: app/controllers/public/index.js (hunk 7 above)
Comment (iamareebjamal) [inline app/controllers/public/index.js:5]: As the review says, you imported the function but never used it

File: app/controllers/public/index.js (hunk 9 above)
Comment (iamareebjamal) [inline app/controllers/public/index.js:139]: There is no RSVP defined and there is no need to change from try catch to then catch

**Comment by iamareebjamal** [comment]
As you can already see, the build is failing

**Comment by EmperorYP7** [comment]
> As you can already see, the build is failing

Tried to implement it again, sir. Hope this'd solve the issues.

File: app/mixins/event-wizard.js [hunk 13]
```diff
@@ -48,7 +48,7 @@ export default Mixin.create(MutableArray, CustomFormMixin, {
   async saveEventData(propsToSave = []) {
     const { event } = this.model;
     const data = {};
-    const results = await Promise.allSettled(propsToSave.map(property => {
+    const results = await RSVP.allSettled(propsToSave.map(property => {
```
Comment (iamareebjamal) [inline app/mixins/event-wizard.js:51]: Return type of RSVP.allSettled is different from Promise.allSettled. Please fix the code accordingly

Comment (EmperorYP7) [reply app/mixins/event-wizard.js:51]: I tried to get around this, sir. Do review it once. Any specific suggestion or resource to guide me through this?

Comment (iamareebjamal) [reply app/mixins/event-wizard.js:51]: Resolve conflicts and I will test the results

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5583663&bid=18300760)

File: app/controllers/public/index.js [hunk 14]
```diff
@@ -2,6 +2,7 @@ import classic from 'ember-classic-decorator';
 import { action, computed } from '@ember/object';
 import Controller from '@ember/controller';
 import { htmlSafe } from '@ember/string';
+import RSVP from 'rsvp';
```
Comment (niranjan94) [inline app/controllers/public/index.js:5]: Codacy found an issue: ['RSVP' is defined but never used.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5583663)

Comment (iamareebjamal) [inline app/controllers/public/index.js:5]:
```suggestion
```

**Comment by EmperorYP7** [comment]
Thank you sir.

**Pull request merged**